tests/kernel/sched/wraparound/src/main.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Akaiwa Wataru <akaiwa@sonas.co.jp>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+static k_tid_t thread_id;
+
+static void alarm_callback(struct k_timer *timer)
+{
+	k_wakeup(thread_id);
+}
+
+K_TIMER_DEFINE(alarm, alarm_callback, NULL);
+
+/**
+ * @brief Test 32-bit tick wraparound during k_sleep() execution
+ */
+ZTEST(wraparound, test_tick_wraparound_in_sleep)
+{
+	static const uint32_t start_ticks = 0xffffff00; /* It wraps around after 256 ticks! */
+	static const uint32_t timeout_ticks = 300;      /* 3 seconds @ 100Hz tick */
+	static const uint32_t wakeup_ticks = 10;        /* 100 ms @ 100Hz tick */
+
+	sys_clock_tick_set(start_ticks);
```

Comment:
Hah, nice.  I didn't know this API existed[1].  Looks like you aren't the first person needing to white-box the tick value, @cfriedt beat you to it in commit 4108e1474038bf877185d1cb5e791d2f5bef38e5.

[1] Yes, the astute will point out that I did in fact review that PR...

Reply:
Yes, I'm lucky.
I will arrange commit order after fixing up the test. Please wait a little longer...